networkx/algorithms/tests/test_walks.py

```diff
@@ -0,0 +1,33 @@
+"""Unit tests for the :mod:`networkx.algorithms.walks` module."""
+
+import pytest
+
+import networkx as nx
+
+pytest.importorskip("numpy")
+pytest.importorskip("scipy")
+
+
+def test_neither_source_nor_target():
+    G = nx.DiGraph([(0, 1), (1, 2), (2, 0)])
+    num_walks = nx.number_of_walks(G, 3)
+    expected = {0: {0: 1, 1: 0, 2: 0}, 1: {0: 0, 1: 1, 2: 0}, 2: {0: 0, 1: 0, 2: 1}}
+    assert num_walks == expected
+
+
+def test_no_target():
+    G = nx.cycle_graph(4)
+    source = 0
+    k = 3
+    num_walks = nx.number_of_walks(G, k)
+    expected = {0: 0, 1: 4, 2: 0, 3: 4}
+    assert num_walks[source] == expected
```

Comment:
This is very much a nit, but we can save a few lines here by getting rid of some of the local variables:

```suggestion
    G = nx.cycle_graph(4)
    num_walks = nx.number_of_walks(G, 3)
    expected = {0: 0, 1: 4, 2: 0, 3: 4}
    assert num_walks[0] == expected
```

Also, in the grander scheme it might be better to do the "complete" test, i.e. set `expected` to be the full dict-of-dicts that maps all nodes to all other nodes, and assert that this is correct for *all* of num_walks, rather than just one instance.